### PR TITLE
Notify third-party admins, take 2

### PR DESCRIPTION
**Pull request by kaapstorm**
##### SUMMARY
This PR responds to feedback on #25452 and adds a few more commits to use the new `notify_error()` method. I've opened a new PR because I rebased off master to try to keep commits clean and readable. :tropical_fish: 

There is some feedback about the standard for names for observables that I haven't responded to ...

> I think that the new standard is to use hungarian notation https://github.com/dimagi/commcare-hq/pull/25394/files

... I will try to implement that in a follow-up PR. The reason I've used Python convention / snake case for observables is because [the template that renders the form](https://github.com/dimagi/commcare-hq/blob/acea6efeaf2484bb92012e79a1917b809bdb5566/corehq/motech/openmrs/templates/openmrs/partials/openmrs_importer_template.html#L39) uses [a Django form](https://github.com/dimagi/commcare-hq/blob/63b71f5b0cfb5f53297f13089dbdeb9a535a49e1/corehq/motech/openmrs/forms.py#L69)'s field's `html_name` property as the name for the observable. That makes the template and the viewmodel tightly coupled to a Django form in a non-obvious way. (It evolved from when a domain could only import from one OpenMRS server. When we allowed for importing from multiple servers, we kept the form, and wrote a template to render the viewmodel as if it was the form. ... It seemed like a good idea at the time :see_no_evil: )

I would like to drop the Django form and rewrite the template. But that feels out of scope of this PR.  (And I would also like to get this deployed before then; there are a couple of projects that would like to use it soon.)

Also worth noting is that the model stores addresses in a `StringProperty`, not a `StringListProperty`. I chose this to keep the view simple. (I tried using `StringListProperty` in the model with [`MultiEmailField`](https://github.com/dimagi/commcare-hq/blob/1b4a2a6c08ca3b50a4e6e86e2c57220a64f32b5c/corehq/apps/hqwebapp/fields.py#L52) in the form, and adding a repeater behaved as expected, but when editing the repeater, the form did not render initial values; the field was always blank).

cc @emord @orangejenny 

##### FEATURE FLAG
Currently "OpenMRS integration", potentially also "DHIS2 integration"

##### PRODUCT DESCRIPTION
Adds the ability to notify third-party administrators of integration errors.

### CODE REVIEW

**Comment by orangejenny** [comment]
> The reason I've used Python convention / snake case for observables...

This sounds reasonable. There are a bunch of places where our js has unconventional naming for similar reasons, like data coming directly from python and then using knockout's automatic mapping functions to make the js object. I don't love this but don't think we have a good all-purpose solution (someday making our naming conventions standard across both python and js?). We don't turn on ESLint's rule about casing for object properties for this reason.

> Also worth noting is that the model stores addresses in a StringProperty...

This is a bit unfortunate, inputs for multiple emails are one of the areas that I'd like to establish a stronger pattern for, both on the front end (use the same js handling) and back end (store them as lists). The select2 upgrade done this year switched select2s from comma-separated strings to lists, so we've been making progress in this general area. That said, we still do this a few different ways, and multi-selects can be really finicky, so I can't blame you for deciding this is out of scope.

**Review by orangejenny — APPROVED** [review]

File: corehq/motech/openmrs/repeater_helpers.py [hunk 1]
```diff
@@ -179,10 +177,6 @@ def check_duplicate_case_match(domain, case_type, external_id):
                    f'external_id="{external_id}" and case type="{case_type}".')
     else: # error == LookupErrors.NotFound:
         return
-    _assert(False, message +
```
[the remark this answers is not in the document]
Comment (kaapstorm) [reply corehq/motech/openmrs/repeater_helpers.py]: I thought so too for a while, but I found that soft_assert isn't like a real assert.

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/motech/openmrs/repeater_helpers.py]: ohhhh, i get it now

**Comment by esoergel** [comment]
It's `DEFAULT_FROM_EMAIL`, not `DEFAULT_FROM_ADDRESS`.  Just pushed a quick fix for that.
```
======================================================================
ERROR: corehq.motech.tests.test_requests:RequestsTests.test_notify_error_address_list
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/motech/tests/test_requests.py", line 137, in test_notify_error_address_list
    requests.notify_error('foo')
  File "/mnt/commcare-hq-ro/corehq/motech/requests.py", line 151, in notify_error
    from_email=settings.DEFAULT_FROM_ADDRESS,
  File "/vendor/lib/python3.6/site-packages/django/conf/__init__.py", line 57, in __getattr__
    val = getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'DEFAULT_FROM_ADDRESS'
```

**Review by millerdev — APPROVED** [review]

**Comment by kaapstorm** [comment]
It looks like I never clicked the "Comment" button last week Thursday :man_facepalming: ... Here's what I wrote: 

Thanks everyone. Thanks for the fix, @esoergel, and for getting this in before deploy. @orangejenny I would like to get `MultiEmailField` working here in the future. It seems to validate initial values; it just doesn't display them. I'll see if I can resolve that.

**Pull request merged**